iceoryx2/tests/service_event_tests.rs

```diff
@@ -545,6 +602,214 @@ mod service_event {
+            assert_that!(id, eq Some(EventId::new(13)));
+        })
+    }
+
+    #[test]
+    fn blocking_wait_blocks_until_notification<Sut: Service>() {
+        wait_blocks_until_notification(|l: &Listener<Sut>| {
+            let id = l.blocking_wait().unwrap();
+            assert_that!(id, eq Some(EventId::new(13)));
+        })
+    }
+
+    #[test]
+    fn try_wait_collects_all_notifications<Sut: Service>() {
+        const NUMBER_OF_NOTIFICATIONS: usize = 8;
+        wait_collects_all_notifications(NUMBER_OF_NOTIFICATIONS, |l: &Listener<Sut>, ids| {
+            while let Some(id) = l.try_wait().unwrap() {
+                assert_that!(ids.insert(id), eq true);
+            }
+        });
+    }
+
+    #[test]
+    fn timed_wait_collects_all_notifications<Sut: Service>() {
+        const NUMBER_OF_NOTIFICATIONS: usize = 8;
+        wait_collects_all_notifications(NUMBER_OF_NOTIFICATIONS, |l: &Listener<Sut>, ids| {
+            for _ in 0..NUMBER_OF_NOTIFICATIONS {
+                let id = l.timed_wait(TIMEOUT).unwrap().unwrap();
+                assert_that!(ids.insert(id), eq true);
+            }
+        });
+    }
+
+    #[test]
+    fn blocking_wait_collects_all_notifications<Sut: Service>() {
+        const NUMBER_OF_NOTIFICATIONS: usize = 8;
+        wait_collects_all_notifications(NUMBER_OF_NOTIFICATIONS, |l: &Listener<Sut>, ids| {
+            for _ in 0..NUMBER_OF_NOTIFICATIONS {
+                let id = l.blocking_wait().unwrap().unwrap();
+                assert_that!(ids.insert(id), eq true);
+            }
+        });
+    }
+
+    #[test]
+    fn try_wait_all_does_not_block<Sut: Service>() {
+        let _watch_dog = Watchdog::new();
+        let service_name = generate_name();
+
+        let sut = Sut::new(&service_name).event().create().unwrap();
+
+        let listener = sut.listener().create().unwrap();
+
+        let mut callback_called = false;
+        assert_that!(listener.try_wait_all(|_| callback_called = true), is_ok);
+        assert_that!(callback_called, eq false);
+    }
+
+    #[test]
+    fn timed_wait_all_blocks_for_at_least_timeout<Sut: Service>() {
+        let _watch_dog = Watchdog::new();
+        let service_name = generate_name();
+
+        let sut = Sut::new(&service_name).event().create().unwrap();
+
+        let listener = sut.listener().create().unwrap();
+
+        let now = Instant::now();
+        let mut callback_called = false;
+        assert_that!(
+            listener.timed_wait_all(|_| callback_called = true, TIMEOUT),
+            is_ok
+        );
+        assert_that!(now.elapsed(), time_at_least TIMEOUT);
+        assert_that!(callback_called, eq false);
+    }
+
+    #[test]
+    fn timed_wait_all_blocks_until_notification<Sut: Service>() {
+        wait_blocks_until_notification(|l: &Listener<Sut>| {
+            assert_that!(
+                l.timed_wait_all(|id| assert_that!(id, eq EventId::new(13)), TIMEOUT * 1000),
+                is_ok
+            );
```

Comment:
From my understanding it could happen that `timed_wait_all` runs into a timeout and does not execute the callback while still returning `Ok(())`. Is this correct?

Reply:
I fixed it